lib/msf/core/web_services/authentication/strategies/user_password.rb

```diff
@@ -16,15 +16,19 @@ class UserPassword < Warden::Strategies::Base
       # Check if request contains valid data and should be authenticated.
       # @return [Boolean] true if strategy should be run for the request; otherwise, false.
       def valid?
-        params['username'] && params['password']
+        body = JSON.parse(request.body.read, symbolize_names: true)
```

Comment:
Should this module `require 'json'` should it be used somewhere JSON has not already been required?